test/javascript/features/config-watcher/config-watcher.test.ts

```diff
@@ -0,0 +1,42 @@
+import path from 'path';
+import fs from 'fs-extra';
+import Scripts from '../../../scripts';
+import { waitForPort } from '../../../utils';
+
+jest.setTimeout(60 * 1000);
+
+const scripts = Scripts.setupProjectFromTemplate({
+  templateDir: __dirname,
+  projectType: 'javascript',
+});
+
+const packageJSON = path.join(scripts.testDirectory, 'package.json');
+
+const updatedPort = 3300;
+
+describe.each(['dev'] as const)('Start command config watcher [%s]', mode => {
+  it('should restart on config change', async () => {
+    await scripts[mode](async () => {
+      //Make sure that yoshi is fully loaded
+      await page.goto(`${scripts.serverUrl}`);
+
+      const data = JSON.parse(fs.readFileSync(packageJSON, 'utf-8'));
+
+      await new Promise(resolve => setTimeout(resolve, 1000));
+
+      data.yoshi.servers.cdn.port = updatedPort;
+      fs.writeFileSync(packageJSON, JSON.stringify(data));
+
+      await waitForPort(updatedPort, { timeout: 20000 });
+      await page.goto('http://localhost:' + updatedPort);
```

Comment:
nit, use the newer syntax:

```suggestion
      await page.goto(`http://localhost: ${updatedPort}`);
```